src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -998,7 +998,9 @@ memory::ptr primitive_inst::allocate_internal_buffer(size_t idx, bool reset) {
         alloc_type = engine.get_lockable_preferred_memory_allocation_type();
     }
     GPU_DEBUG_LOG << "=> allocate to " << alloc_type << std::endl;
-    return engine.allocate_memory(layout, alloc_type, reset);
+
+    // Reuse intermediate buffer like output buffer.
+    return _network.get_memory_pool().get_intermediate_memory(layout, _node->id(), _network.get_id(), _node->get_memory_dependencies(), alloc_type);
```

Comment:
There is `bool reset` function parameter which is supposed to reset memory if needed, so I think we need to handle it to avoid misunderstanding and problebs in the future
Also, `network::allocate_primitives()` uses `prim->allocate_internal_buffers()` with default `true` value

Reply:
`memory_pool::get_memory()` is used to allocate memory at `prim->allocate_internal_buffer(..)`. So, I remained reset argument.